modules/planning/open_space/trajectory_smoother/dual_variable_warm_start_osqp_interface_test.cc

```diff
@@ -85,16 +92,87 @@ TEST_F(DualVariableWarmStartOSQPInterfaceTest, optimize) {
 
   bool res = ptop_->optimize();
   EXPECT_TRUE(res);
-
   ptop_->get_optimization_results(&l_warm_up, &n_warm_up);
+
+  // regard ipopt_qp as ground truth result
+  Eigen::MatrixXd l_warm_up_gt(obstacles_edges_sum, horizon_ + 1);
+  Eigen::MatrixXd n_warm_up_gt(4 * obstacles_num_, horizon_ + 1);
+
+  Ipopt::SmartPtr<Ipopt::TNLP> problem = pt_gt_;
+  // Create an instance of the IpoptApplication
+  Ipopt::SmartPtr<Ipopt::IpoptApplication> app = IpoptApplicationFactory();
+
+  app->Options()->SetIntegerValue(
+      "print_level",
+      planner_open_space_config_.dual_variable_warm_start_config()
+          .ipopt_config()
+          .ipopt_print_level());
+  app->Options()->SetIntegerValue(
+      "mumps_mem_percent",
+      planner_open_space_config_.dual_variable_warm_start_config()
+          .ipopt_config()
+          .mumps_mem_percent());
+  app->Options()->SetNumericValue(
+      "mumps_pivtol",
+      planner_open_space_config_.dual_variable_warm_start_config()
+          .ipopt_config()
+          .mumps_pivtol());
+  app->Options()->SetIntegerValue(
+      "max_iter", planner_open_space_config_.dual_variable_warm_start_config()
+          .ipopt_config()
+          .ipopt_max_iter());
+  app->Options()->SetNumericValue(
+      "tol", planner_open_space_config_.dual_variable_warm_start_config()
+          .ipopt_config()
+          .ipopt_tol());
+  app->Options()->SetNumericValue(
+      "acceptable_constr_viol_tol",
+      planner_open_space_config_.dual_variable_warm_start_config()
+          .ipopt_config()
+          .ipopt_acceptable_constr_viol_tol());
+  app->Options()->SetNumericValue(
+      "min_hessian_perturbation",
+      planner_open_space_config_.dual_variable_warm_start_config()
+          .ipopt_config()
+          .ipopt_min_hessian_perturbation());
+  app->Options()->SetNumericValue(
+      "jacobian_regularization_value",
+      planner_open_space_config_.dual_variable_warm_start_config()
+          .ipopt_config()
+          .ipopt_jacobian_regularization_value());
+  app->Options()->SetStringValue(
+      "print_timing_statistics",
+      planner_open_space_config_.dual_variable_warm_start_config()
+          .ipopt_config()
+          .ipopt_print_timing_statistics());
+  app->Options()->SetStringValue(
+      "alpha_for_y",
+      planner_open_space_config_.dual_variable_warm_start_config()
+          .ipopt_config()
+          .ipopt_alpha_for_y());
+  app->Options()->SetStringValue(
+      "recalc_y", planner_open_space_config_.dual_variable_warm_start_config()
+          .ipopt_config()
+          .ipopt_recalc_y());
```

Comment:
take ipopt_config() on stack, shorten the code

Reply:
> take ipopt_config() on stack, shorten the code

fixed